change meter/group selection logic and dropdown sorting

spearec opened the pull request:
# Description

This PR changes the logic for determining when a meter/group is selected or deselected and changes the sorting for the dropdown menus. The new sorting has a divider to clearly show the user why incompatible meters are greyed out, as well as putting meters without GPS below those with GPS in the map view while preserving the original order.

![image](https://user-images.githubusercontent.com/113216142/204296522-3346a3c2-d8dd-42a3-a75b-f6f1ab42ee21.png)

Fixes #600 

## Type of change

- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

spearec committed:
change way in which meters are disabled

spearec committed:
Merge branch 'OpenEnergyDashboard:development' into mapSelection

spearec committed:
new way to determine meter selection

spearec committed:
Revert "change way in which meters are disabled"

This reverts commit e3f87e3cac95e0edefa1d582aff960ea7fc7ffe2.

spearec committed:
rename variables and copy logic to groups

spearec committed:
Merge branch 'mapSelection' of https://github.com/spearec/OED into mapSelection

spearec committed:
change selection dropdown sorting and add divider

huss commented on src/client/app/components/ChartDataSelectComponent.tsx:
```diff
@@ -225,10 +291,24 @@ export default function ChartDataSelectComponent() {
 					placeholder={intl.formatMessage(messages.selectMeters)}
-					onValuesChange={(newSelectedMeterOptions: SelectOption[]) =>
-						dispatch(changeSelectedMeters(newSelectedMeterOptions.map(s => s.value)))}
+					// change logic for selection: compute diff between dataProps.selectedMeters and newSelectedMeterOptions\
```
Remove \ at end of comment?

huss commented on src/client/app/components/ChartDataSelectComponent.tsx:
```diff
@@ -212,10 +266,22 @@ export default function ChartDataSelectComponent() {
-					onValuesChange={(newSelectedGroupOptions: SelectOption[]) =>
-						dispatch(changeSelectedGroups(newSelectedGroupOptions.map(s => s.value)))}
+					onValuesChange={(newSelectedGroupOptions: SelectOption[]) => {
+						const allSelectedGroupIDs: number[] = dataProps.allSelectedGroups.map(s => s.value);
```
There are more comments in the section below but even with these I think some more would help future developers. Could you describe why each of the filters work and any other choices?

huss reviewed (changes requested):
Thanks to @spearec for more good work. I like the separators on the menus and the consistent use. I have a couple of comments on commenting but nothing else. Testing indicates it works as expected.

spearec committed:
commenting chartDataSelectComponent

spearec commented:
I've added some more comments and changed variable names to be consistent with term usage. This makes the distinction between displayable items (an admin-level setting) and compatible items (determined by graph type/unit selection).

huss commented:
Everything looks good. Thanks.
